### DISCUSSION TITLE
Metric aggregation and top_count_keys

### DISCUSSION THREAD

**Discussion by markus-nclose** [jertel/elastalert2#862] [Q&A] [answered]
HI All, 

I have a strange issue I'm facing. My Elastic has non-analyzed fields saved as .raw and not .keyword. 
I'm trying to include more fields in my metric aggregation rule by means of ```top_count_keys``` as per documentation. ```.keyword``` is appened to these fields and returns no results.

I tried using the ```raw_count_keys``` option, but it still shows ```.keyword``` as the field. 

Am I missing something obvious?

Here is the rule for reference

```
type: metric_aggregation

metric_agg_key: src_port.raw
metric_agg_type: cardinality
doc_type: _doc

query_key: dst_ip.raw
max_threshold: 1000

top_count_keys: [dst_ip.raw, Image, computer_name]
raw_count_keys: true
```

Thanks.
Markus

**Comment by markus-nclose** [comment]
I tried manually adding the .raw to the field to no avail.

Just to add output:

Once the rule detects activity, the output shows:

```
Included term dst_ip.raw may be missing or null
top_count_key dst_ip.raw may be missing
```
Followed by:

```
dst_ip.raw.keyword:
No events found.

Image.keyword:
No events found.

computer_name.keyword:
No events found.

```

**Comment by jertel** [comment, marked as answer]
From the docs:
![image](https://user-images.githubusercontent.com/16639611/169810426-39a6ed89-b6ce-435a-bd99-d84207b9cb2b.png)

ElastAlert 2 no longer supports ES versions earlier than ES 7, so this documentation needs some work, but essentially it's saying that as of ES 5 (and up) it appends ".keyword" instead of ".raw".

Have you tried
```
top_count_keys: [dst_ip.raw, Image.raw, computer_name.raw]
raw_count_keys: false
```
?

**Reply by markus-nclose** [reply]
Aah that makes more sense. We have a strange mapping template in our ES 7.15 for the raw fields. 

I thought I tried it, but clearly not. Seems be loading the fields correctly now but still not getting events for them - that might be a separate issue all together. 

```
dst_ip.raw:
No events found.

Image.raw:
No events found.

computer_name.raw:
No events found.
```

Thanks for the help.
